contentcuration/kolibri_public/management/commands/rectify_incorrect_contentnode_source_fields.py

```diff
@@ -0,0 +1,134 @@
+                modified__lt=filter_date
+                # published=True,
+            )
+        ).annotate(
+            coalesced_provider=Coalesce("provider", Value("")),
+            coalesced_author=Coalesce("author", Value("")),
+            coalesced_aggregator=Coalesce("aggregator", Value("")),
+            coalesced_license_id=Coalesce("license_id", -1),
+        )
+
+        diff_combined = diff.annotate(
+            original_source_node_f_changed=Exists(
+                original_source_nodes.filter(
+                    ~Q(coalesced_provider=OuterRef("coalesced_provider"))
+                    | ~Q(coalesced_author=OuterRef("coalesced_author"))
+                    | ~Q(coalesced_aggregator=OuterRef("coalesced_aggregator"))
+                    | ~Q(coalesced_license_id=OuterRef("coalesced_license_id"))
+                )
+            )
+        ).filter(original_source_node_f_changed=True)
+
+        final_nodes = diff_combined.values(
+            "id",
+            "original_channel_id",
+            "original_source_node_id",
+            "coalesced_provider",
+            "coalesced_author",
+            "coalesced_aggregator",
+            "coalesced_license_id",
+            "original_source_node_f_changed",
+        ).order_by()
+
+        for item in final_nodes:
+            base_node = ContentNode.objects.get(pk=item["id"])
+
+            original_source_channel_id = item["original_channel_id"]
+            original_source_node_id = item["original_source_node_id"]
+            tree_id = (
+                Channel.objects.filter(pk=original_source_channel_id)
+                .values_list("main_tree__tree_id", flat=True)
+                .get()
+            )
+            original_source_node = ContentNode.objects.filter(
+                tree_id=tree_id, node_id=original_source_node_id
+            )
+
+            base_channel = Channel.objects.get(pk=item['channel_id'])
+
+            to_be_republished = not (base_channel.main_tree.get_family().filter(changed=True).exists())
+
+            if original_source_channel_id is not None and original_source_node.exists():
+                # original source node exists and its source fields dont match
+                # update the base node
+                if base_node[0].author != original_source_node[0].author:
+                    base_node[0].author = original_source_node[0].author
+                if base_node[0].provider != original_source_node[0].provider:
+                    base_node[0].provider = original_source_node[0].provider
+                if base_node[0].aggregator != original_source_node[0].aggregator:
+                    base_node[0].aggregator = original_source_node[0].aggregator
+                if base_node[0].license != original_source_node[0].license:
+                    base_node[0].license = original_source_node[0].license
+                base_node[0].save()
+
+                if to_be_republished and base_channel.public:
```

Comment:
whats the best way to republish this?

Reply:
Instead of `base_channel.public` I believe we want `base_channel.last_published is not None`